### DISCUSSION TITLE
Using xpath_query with xpath_variable_set

### DISCUSSION THREAD

**Discussion by CisimEjirtimid** [zeux/pugixml#430] [Q&A] [answered]
Hello, I'm having an issue with using `xpath_query` with the query being something along the lines of
`//node[@$attributeName = $attributeValue]`
with `attributeName` and `attributeValue` being variables I want to parametrize through use of `xpath_variable_set`.
But when I set both variables to be of string type, I only get an exception saying that it's `Unrecognized node test`.
What am I missing here?

**Comment by zeux** [comment, marked as answer]
Variables can't be used in @-tests. You're looking for something like `//node[attribute::*[name() = $attributeName and value() = $attributeValue]]`

**Reply by CisimEjirtimid** [reply]
Hey, thanks for quick response!
The solution ended up being a bit different, but you certainly helped me with correctly setting up the query.
Anyway, the right query is `//node[@*[name() = $attributeName] = $attributeValue]`.

Thank you once more!
